### PR TITLE
Add support for FreeRTOS ported by Espressif

### PR DESCRIPTION
**Pull request by HanesSciarrone**
### Description of the feature or fix

The FreeRTOS ported by Espressif changes the standard FreeRTOS API to support multi-core SoCs. Therefore, this change generate compatibility with the new Espressif API. 

Fixes issue #5810

### CODE REVIEW

File: Kconfig [hunk 1]
```diff
@@ -135,12 +135,14 @@ menu "LVGL configuration"
 				bool "1: PTHREAD"
 			config LV_OS_FREERTOS
 				bool "2: FREERTOS"
+            config LV_OS_FREERTOS_ESP
```
Comment (kisvegabor) [inline Kconfig]: Check the indentation.

**Review by kisvegabor — COMMENTED** [review]
Thank you! 

It just came to my mind that we could automatically detect the ESP environment via an `#ifdef`. E.g. `#ifdef ESP_PLATFORM` I'm not sure if really `ESP_PLATFORM` shall  be used, but we are using it in [Cmake](https://github.com/lvgl/lvgl/blob/master/CMakeLists.txt#L11C4-L11C16).

What do you think about it?

File: Kconfig [hunk 2]
```diff
@@ -150,11 +152,12 @@ menu "LVGL configuration"
 			default 0 if LV_OS_NONE
 			default 1 if LV_OS_PTHREAD
 			default 2 if LV_OS_FREERTOS
-			default 3 if LV_OS_CMSIS_RTOS2
-			default 4 if LV_OS_RTTHREAD
-			default 5 if LV_OS_WINDOWS
+            default 3 if LV_OS_FREERTOS_ESP
```
Comment (PGNetHun) [inline Kconfig]: Use same indentation (TAB) in your new lines.

**Review by PGNetHun — CHANGES_REQUESTED** [review]
Just minor indentation corrections are needed, the config and code changes seem OK.

File: src/osal/lv_freertos.c [hunk 3]
```diff
@@ -336,8 +345,11 @@ static void prvCheckMutexInit(lv_mutex_t * pxMutex)
     if(pxMutex->xIsInitialized == pdFALSE) {
         /* Mutex initialization must be in a critical section to prevent two threads
          * from initializing it at the same time. */
+#if LV_USE_OS == LV_OS_FREERTOS
         taskENTER_CRITICAL();
-
+#elif LV_USE_OS == LV_OS_FREERTOS_ESP
+        taskENTER_CRITICAL(&critSectionMux);
+#endif
```
Comment (XuNeo) [inline src/osal/lv_freertos.c]: How about define a macro above to handle differences between FREERTOS and FREERTOS_ESP?

```c
#if LV_USE_OS == LV_OS_FREERTOS
        #define _enter_critical() /*need better name*/ taskENTER_CRITICAL();
        #define _exit_critical()
#elif LV_USE_OS == LV_OS_FREERTOS_ESP
        #define _enter_critical() taskENTER_CRITICAL(&critSectionMux);
        #define _exit_critical()
#endif
```

**Comment by HanesSciarrone** [comment]
Hi @kisvegabor 

I've done some research on the ESP_PLATFORM macro. As far as I can see the macro is set to 1 when the CMake file is processed within the ESP-IDF build system [Documentation ESP-IDF](https://docs.espressif.com/projects/esp-idf/en/stable/esp32/api-guides/build-system.html#preset-component-variables). My doubt is if we use the Arduino framework instead of ESP-IDF, I'll test it in my development environment while writing in the issue to ask the users if they can test it also.

**Comment by HanesSciarrone** [comment]
Hi @kisvegabor 

I've tested the approach you commented using the ESP_PLATFORM macro and it works perfectly. I think that it is a better approach because the modified files are less.

**Comment by kisvegabor** [comment]
Great! Please update PR accordingly.

**Review by PGNetHun — APPROVED** [review]
Great, these new changes are even better!

**Comment by HanesSciarrone** [comment]
@kisvegabor What do you mean by updating the PR accordingly? I already uploaded the changes when I replied to you.

**Review by XuNeo — APPROVED** [review]

**Review by kisvegabor — APPROVED** [review]
My bad, all look good! Thank you!

**Pull request merged**